helix-core/src/test/java/org/apache/helix/integration/rebalancer/TestStickyRebalanceStrategy.java

```diff
@@ -147,9 +140,27 @@ public void afterTest() throws InterruptedException {
     _clusterVerifier.verifyByPolling();
   }
 
+  @Test
+  public void testNoSameZoneAssignment() throws Exception {
+    setTopologyAwareAndGlobalMaxPartitionAllowedPerInstanceInCluster(CLUSTER_NAME, 1);
+    Map<String, ExternalView> externalViews = createTestDBs();
```

Comment:
Does `createTestDBs` use the verifier to make sure the controller thread is done computing the assignment before starting the assertions? If not, this should be added.

Reply:
Do you mean by following?
```
Assert.assertTrue(_clusterVerifier.verifyByPolling());
```
If yes, then yes and here is the code pointer: https://github.com/apache/helix/blob/helix-gateway-service/helix-core/src/test/java/org/apache/helix/integration/rebalancer/TestStickyRebalanceStrategy.java#L332